### DISCUSSION TITLE
Automating Reddit website using Cypress

### DISCUSSION THREAD

**Discussion by sinirajappan** [cypress-io/cypress#24351] [Questions and Help] [answered]
Has anyone tried to automate www.reddit.com using cypress 10 as part of any test project? When I tried to invoke the above url using cy.visit, for some reason the Test Runner window closes on its own not showing what happened to my tests. But when I change the resolution to that of a mobile device, it works just fine and I can see my test passes. www.reddit.com/login also loads fine, however after entering username/password and clicking login, the homepage does not load after login as it does when manually done.
Does Reddit web version have elements that Cypress 10 cannot work with?

**Comment by filiphric** [comment, marked as answer]
I’m thinking you are trying to use Cypress as a general-purpose automation tool and not a testing tool. Big sites like gmail, reddit, facebook have different security measures in place that keep them from being opened in iframe (as Cypress does). This means that reddit probably kicks you out of its page the minute you try to access it via iframe.

Cypress is a testing tool, which means that it is built for testing an application you are somehow involved with, either as a developer or a tester. It will not work for scraping the web or performing automation scripts on a third party site as this is not something it was built for.

[more explanation in this doc FAQ](https://docs.cypress.io/faq/questions/general-questions-faq#Can-I-use-Cypress-to-script-user-actions-on-an-external-site-like-gmail-com)

**Reply by sinirajappan** [reply]
That quite makes sense now. Thank you much for your reply!
